biz.aQute.launcher/src/aQute/launcher/plugin/ProjectLauncherImpl.java

```diff
@@ -152,7 +155,7 @@ public void cleanup() {
 
 	@Override
 	public String getMainTypeName() {
-		return EMBEDDED_LAUNCHER;
+		return getProject().get(MAIN_CLASS, EMBEDDED_LAUNCHER);
```

Comment:
Looks good. Can you squash all the commits in to a single commit including changing the above line to

```java
getProject().getProperty(MAIN_CLASS, EMBEDDED_LAUNCHER);
```

`get` is the Domain method which just delegates to the Processor `getProperty` method anyway.

Reply:
Done